### ISSUE TITLE
Auto-open code needs to be in separate function

### ISSUE THREAD

**Issue by droidmonkey** [keepassxreboot/keepassxc#1259]
The auto-open code that was approved in PR #1053 was incorrectly stuffed into a totally unrelated function. This code needs to be placed in it's own function and called appropriately.

https://github.com/keepassxreboot/keepassxc/blob/develop/src/gui/DatabaseTabWidget.cpp#L628

**Comment by TheZ3ro** [comment]
Also, add to auto-open urls that start with `kdbx://`, now only `file://` is supported but KeePass plugin also use the first one

**Comment by TheZ3ro** [comment]
Related #1420. The autoopen function should run in background and not open the Main Window

**Comment by droidmonkey** [comment]
The setup and configuration of auto open should be added as a GUI element to the Database Settings.

**Comment by droidmonkey** [comment]
This was fixed with #2504
